### PR TITLE
Add Nutanix platform

### PR DESCRIPTION
**Pull request by thunderboltsid**
Set Nutanix Load balancer annotation to nil.

### CODE REVIEW

**Comment by rfredette** [comment]
This change makes sense to me, but I ran `make test` against this branch and I'm getting an error that `configv1.NutanixPlatformType` is undefined. It looks like openshift/api has been updated to include Nutanix as a platform type, so can you update the vendored version of openshift/api? You should just need to run 
```
go get github.com/openshift/api@master
go mod vendor
make generate
```
then commit the updated files.

That said, I'm going to ask you to update openshift/api again once your API PR merges, so if you want to hold off until that's done, I'm fine with that.

**Comment by thunderboltsid** [comment]
> This change makes sense to me, but I ran `make test` against this branch and I'm getting an error that `configv1.NutanixPlatformType` is undefined. It looks like openshift/api has been updated to include Nutanix as a platform type, so can you update the vendored version of openshift/api? You should just need to run
> 
> ```
> go get github.com/openshift/api@master
> go mod vendor
> make generate
> ```
> 
> then commit the updated files.
> 
> That said, I'm going to ask you to update openshift/api again once your API PR merges, so if you want to hold off until that's done, I'm fine with that.

seems like I just had to rebase the commit in https://github.com/openshift/cluster-ingress-operator/pull/730 to the latest master (API dependency on the current master already has the nutanix platform type) and now the tests should pass on the branch.

**Comment by rfredette** [comment]
/lgtm
thanks!

File: test/e2e/configurable_route_test.go [hunk 1]
```diff
@@ -369,6 +373,7 @@ func TestConfigurableRouteNoConsumingUserNoRBAC(t *testing.T) {
 				RelatedObjects:   []configv1.ObjectReference{{}},
 			},
 		},
+		DefaultPlacement: configv1.DefaultPlacementControlPlane,
 	}
 	if err := eventuallyUpdateIngressStatus(t, ingress.Status); err != nil {
 		t.Errorf("error updating ingress.status: %v", err)
```
Comment (Miciah) [inline test/e2e/configurable_route_test.go:379]: You shouldn't need to specify `DefaultPlacement` now that https://github.com/openshift/cluster-config-operator/pull/243 has merged.

File: test/e2e/operator_test.go [hunk 2]
```diff
@@ -2632,6 +2638,9 @@ func newHostNetworkController(name types.NamespacedName, domain string) *operato
 			EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
 				Type: operatorv1.HostNetworkStrategyType,
 			},
+			NodePlacement: operatorv1.NodePlacement{
+				NodeSelector: &selector,
```
Comment (Miciah) [inline test/e2e/operator_test.go:2642]: Why is this needed?

Comment (thunderboltsid) [reply test/e2e/operator_test.go:2642]: I was trying to fix some tests that were failing due to unrelated changes in API. I will remove the commit.

File: test/e2e/configurable_route_test.go (hunk 1 above)
Comment (thunderboltsid) [reply test/e2e/configurable_route_test.go]: Cool, removed the last commit.
